apps/cowswap-frontend/src/common/pure/ClickableAddress/index.tsx

```diff
@@ -36,26 +17,44 @@ export type ClickableAddressProps = {
 export function ClickableAddress(props: ClickableAddressProps) {
   const { address, chainId } = props
 
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  const isMobile = useMediaQuery(Media.upToMedium(false))
+
+  const [openTooltip, setOpenTooltip] = useState(false)
+
   const shortAddress = shortenAddress(address)
 
-  const target = getTarget(address, chainId)
-  const title = getTitle(address, chainId)
+  const target = getExplorerLink(chainId, address, ExplorerDataType.TOKEN)
+
+  const shouldShowAddress = target && !getIsNativeToken(chainId, address)
 
-  const shouldShowAddress = target && title && !isNativeToken(address, chainId)
+  const handleClick = useCallback((event: MouseEvent<HTMLButtonElement>) => {
+    event.stopPropagation?.()
+    setOpenTooltip((prev) => !prev)
+  }, [])
```

Comment:
ubernit: just curious, is `useCallback` saving a lot of work here?  not sure these are worth it

Reply:
cc @cowdan